examples/src/main/java/io/grpc/examples/experimental/CompressingHelloWorldClient.java

```diff
@@ -0,0 +1,116 @@
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ *    * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.grpc.examples.experimental;
+
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ClientInterceptor;
+import io.grpc.Codec;
+import io.grpc.ManagedChannel;
+import io.grpc.MethodDescriptor;
+import io.grpc.examples.helloworld.GreeterGrpc;
+import io.grpc.examples.helloworld.HelloRequest;
+import io.grpc.examples.helloworld.HelloResponse;
+import io.grpc.netty.NegotiationType;
+import io.grpc.netty.NettyChannelBuilder;
+
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * A simple client that requests a greeting from the
+ *      {@link io.grpc.examples.helloworld.HelloWorldServer}.
+ *
+ * <p>This class should act a a drop in replacement for
+ *      {@link io.grpc.examples.helloworld.HelloWorldClient}.
+ */
+public class CompressingHelloWorldClient {
+  private static final Logger logger =
+      Logger.getLogger(CompressingHelloWorldClient.class.getName());
+
+  private final ManagedChannel channel;
+  private final GreeterGrpc.GreeterBlockingStub blockingStub;
+
+  /** Construct client connecting to HelloWorld server at {@code host:port}. */
+  public CompressingHelloWorldClient(String host, int port) {
+    channel =
+        NettyChannelBuilder.forAddress(host, port).negotiationType(NegotiationType.PLAINTEXT)
+            .intercept(new ClientInterceptor() {
+              @Override
+              public <RequestT, ResponseT> ClientCall<RequestT, ResponseT> interceptCall(
+                  MethodDescriptor<RequestT, ResponseT> method, CallOptions callOptions,
+                  Channel next) {
+                return next.newCall(method, callOptions.withCompressor(new Codec.Gzip()));
+              }
+            })
+            .build();
+    blockingStub = GreeterGrpc.newBlockingStub(channel);
```

Comment:
I would have expected `blockingStub.withCompressor(new Codec.Gzip())` in the hello world, instead of using an interceptor. Either here or in `greet()`.